pipeline/train/train.sh

```diff
@@ -30,42 +35,88 @@ mkdir -p "${model_dir}/tmp"
 
 all_model_metrics=(chrf ce-mean-words bleu-detok)
 
-echo "### Training ${model_dir}"
+echo "### Preparing tsv datasets and config"
+
+# Generate a new OpusTrainer config based on a template to fill paths of the datasets
+new_config="${model_dir}/config.opustrainer.yml"
+cp "configs/opustrainer/${model_type}.yml" "${new_config}"
+
+# Iterate over the training sets
+# split the input string into an array
+IFS=',' read -ra datasets <<< "${train_set_prefixes}"
+IFS=',' read -ra alns <<< "${alignments}"
+# loop through the array and get both value and index
+for index in "${!datasets[@]}"; do
```

Comment:
Question: Can you explain what this line and the next mean? My bash isn't good enough to understand these sections.

I also can't grok what exactly `IFS=',' read -ra datasets <<< "${train_set_prefixes}"` is doing here.

Reply:
It just expands the array to generate the indices to iterate on

Reply:
and the next one just element access by index